analyzer/src/main/kotlin/managers/PhpComposer.kt

```diff
@@ -256,6 +265,36 @@ class PhpComposer(analyzerConfig: AnalyzerConfiguration, repoConfig: RepositoryC
         return packages
     }
 
+    /**
+     * Get all names of "virtual" (replaced or provided) packages in the package or lock file.
+     *
+     * While Composer also takes the versions of the virtual packages into account, we simply use priorities here. Since
+     * Composer can't handle the same package in multiple version, we can assume that as soon as a package is found in
+     * composer.lock we can ignore any virtual package with the same name. Since the code later depends on the virtual
+     * packages not accidentally containing a package which is actually installed, we make sure to only return virtual
+     * packages for which are not in the installed package map.
+     */
+    private fun parseVirtualPackageNames(packages: Map<String, Package>, manifest: JsonNode, lockFile: JsonNode)
+            : Set<String> {
+        val replacedNames = HashSet<String>()
+        // The contents of the manifest file, which can also define replacements, is not included in the lock file, so 
```

Comment:
Style nit: We prefer to have a blank line before a comment block to visually group the comment with the code it belongs to. And eventually also add a blank line after the code the comments belongs to.